app/code/community/Bolt/Boltpay/Model/FeatureSwitch.php

```diff
@@ -24,13 +24,23 @@
  */
 class Bolt_Boltpay_Model_FeatureSwitch extends Bolt_Boltpay_Model_Abstract
 {
+    /**
+     * Set flag to true when bolt plugin is updated and we need to update features
+     */
+    private $needUpdateFeatureSwitches = false;
+
     /**
      * This method gets feature switches from Bolt and updates the local DB with
      * the latest values. To be used in upgrade data and webhooks.
      */
-    public function updateSwitchesFromBolt()
+    public function updateFeatureSwitchesFromBolt()
     {
-        $switchesResponse = $this->boltHelper()->getFeatureSwitches();
+        try {
+            $switchesResponse = $this->boltHelper()->getFeatureSwitches();
+        } catch (\Exception $e) {
+            // We already created bugsnag about exception
+            return;
+        }
         $switchesData = @$switchesResponse->data->plugin->features;
 
         if (!is_array($switchesData) || count($switchesData) == 0) {
```

Comment:
I'm curious as to what the possible data types can be.  I would imagine `null` or `array`.  If that is the case, it is cleaner and clearer just to write
```
if ( empty($switchesData) ) {
```

Reply:
Fixed in https://github.com/BoltApp/bolt-magento1/pull/662